### PR TITLE
move Make,Delete,Head,Heal bucket calls to websockets

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
move to Make,Delete,Head,Heal bucket calls to WebSockets

## Motivation and Context
just optimization for relatively very
fewer data transfer calls, but heavier
on the network.

## How to test this PR?
you would need a large enough cluster

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by klauspost** [comment]
https://github.com/minio/minio/pull/18952 Fixes the reported race

**Review by klauspost — APPROVED** [review]
LGTM

File: cmd/peer-s3-client.go [hunk 1]
```diff
@@ -404,7 +423,7 @@ func (sys *S3PeerSys) MakeBucket(ctx context.Context, bucket string, opts MakeBu
 		}, idx)
 	}
 	errs := g.Wait()
-
+
```
Comment (fwessels) [inline cmd/peer-s3-client.go]: Spurious whitespace?

File: cmd/peer-rest-server.go [hunk 2]
```diff
@@ -1421,6 +1421,89 @@ func (s *peerRESTServer) NetSpeedTestHandler(w http.ResponseWriter, r *http.Requ
...
+	if err != nil {
+		return np, grid.NewRemoteErr(err)
+	}
+
+	return np, nil
+}
+
+var headBucketHandler = grid.NewSingleHandler[*grid.MSS, *VolInfo](grid.HandlerHeadBucket, grid.NewMSS, func() *VolInfo { return &VolInfo{} })
+
+// HeadBucketHandler implements peer BuckeInfo call, returns bucket create date.
```
Comment (fwessels) [inline cmd/peer-rest-server.go:1445]: BucketInfo

File: cmd/endpoint.go [hunk 3]
```diff
@@ -411,6 +411,47 @@ func (l EndpointServerPools) GridHosts() (gridHosts []string, gridLocal string)
 	return gridHosts, gridLocal
 }
 
+// FindGridHostsFromPeerPool will return a matching peerPool from provided peer (as string)
```
Comment (fwessels) [inline cmd/endpoint.go:414]: not `(as string)` for this function.

**Review by fwessels — APPROVED** [review]
LGTM

**Pull request merged**